javaslang-benchmark/src/test/java/javaslang/benchmark/MicroBenchmark.java

```diff
@@ -1,30 +1,19 @@
 package javaslang.benchmark;
 
-import javaslang.collection.List;
-import javaslang.collection.Queue;
-import javaslang.collection.Stream;
-import javaslang.collection.Vector;
+import javaslang.collection.*;
 import org.junit.Test;
 import org.openjdk.jmh.annotations.*;
-import org.openjdk.jmh.infra.Blackhole;
 import org.openjdk.jmh.runner.Runner;
-import org.openjdk.jmh.runner.options.Options;
-import org.openjdk.jmh.runner.options.OptionsBuilder;
-import org.openjdk.jmh.runner.options.TimeValue;
+import org.openjdk.jmh.runner.options.*;
 
-
-import java.util.Random;
 import java.util.concurrent.TimeUnit;
 
 @State(Scope.Benchmark)
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 @BenchmarkMode(Mode.Throughput)
 public class MicroBenchmark {
```

Comment:
What is the performance of the current implementation vs `java.util.PriorityQuote` after fixing "comparator instanceof SerializableComparator"?  Any methods in particular?

By the way, you are calling the `PriorityQueueBase#with` method very frequently, which in turns ends up creating a new priority queue object using the the `PriorityQueueBase` constructor, which in turns always casts the `comparator` to `SerializableComparator`. This operation is unnecessary when you call `PriorityQueueBase#with` internally, as you know the comparator is already a `SerializableComparator`. It might be worth overloading the constructor.